### PR TITLE
Audio decoder Jupyter NoteBook

### PR DESCRIPTION
**Pull request by swetha097**
This PR contains a Jupyter Notebook example to showcase a simple audio decoder example.

### CODE REVIEW

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 1]
```diff
@@ -0,0 +1,486 @@
+{
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:1]: @swetha097  : Please remove all the output data from this PR. The user is supposed to run the notebook and get output

Comment (swetha097) [reply docs/examples/notebooks/audio_decoder.ipynb:1]: Done - removed all the cell outputs

**Comment by swetha097** [comment]
> @swetha097 Please address the review comments

Addressed the review comment

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "30b7205e-8719-4b16-8839-7b52c435d0da",
+   "metadata": {},
+   "source": [
+    "This is a simple example show-casing how to set up a simple audio decoder pipeline.\n",
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:8]: change show-casing to demonstrating

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    "\n",
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:9]: remove the empty line and combine with previous sentence into a paragraph

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    "We load and decode audio data using rocAL.\n",
+    "\n",
+    "The input data we will be using to show-case this example is a speech dataset which is present in the form of wav files."
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:12]: remove show-case (there is no hyphen). Change to "input data used for this example is a sample speech dataset available in .. as .wav file)

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4225a5b0-85ba-4d65-85b0-a276f82c177d",
+   "metadata": {},
+   "source": [
+    "We will start by importing rocAL and other necessary packages which is show in the cell below"
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:20]: change to " The following python packages are required to run this example"

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    "import amd.rocal.types as types\n",
+    "import math\n",
+    "import sys\n",
+    "import cv2\n",
+    "import matplotlib.pyplot as plt\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "code",
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:57]: Add explanation for draw_patches

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    "    image = img.detach().numpy()\n",
+    "    audio_data = image.flatten()\n",
+    "    label = idx.cpu().detach().numpy()\n",
+    "    print(\"label: \", label)\n",
+    "    plt.plot(audio_data)\n",
+    "    plt.show()\n",
+    "    plt.close()"
+   ]
+  },
+  {
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:73]: Add explanation for rocal_data_path

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    "\n",
+    "rocal_audio_data_path = os.path.join(rocal_data_path, \"rocal_data\", \"audio\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2cb7c80e-fa78-4fdb-993e-f1f0eb4d62ba",
+   "metadata": {},
+   "source": [
+    "rocAL Pipeline configration with the batch_size, num_threads and weather cpu / gpu backend is used"
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:98]: be more specific. explain batch_size=1, num_threads=8 etc

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    "audio_pipeline.build()\n",
+    "audioIterator = ROCALAudioIterator(audio_pipeline)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1bb26e37-3d99-4da4-92f6-004cb4de8c6c",
+   "metadata": {},
+   "source": [
+    "The output from the iterator contains the audio data as torch tensors, the label of the corresponding data and region of intrest."
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:157]: add coma after data

**Review by rrawther — CHANGES_REQUESTED** [review]
Added a lot of editing changes.
Please make sure clear steps for all Jupyter notebooks as this will be part of rocAL formal documentation. You can use Grammarly for spell and language check if needed

**Comment by sbavasab** [comment]
Addressed the comments

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "df054e1f-9b06-44f4-9aec-80c8798e3225",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install opencv-python\n",
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:26]: how about export ROCAL_DATA_PATH which is needed for running this sample

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+   "id": "d59b9f7a-c997-4155-a9dd-81adf9c31f81",
+   "metadata": {},
+   "source": [
+    "Reading Audio and Labels: The fn.readers.file function reads audio files and their labels from the provided file list.\n",
+    "\n",
+    "Decoding Audio: The fn.decoders.audio function decodes the audio data with specified parameters.\n",
+    "\n",
+    "Parameters:\n",
+    "\n",
+    "audio: The audio data to be decoded.\n",
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:140]: use more readable language for explaining parameters. 
https://github.com/NVIDIA/DALI/blob/main/docs/examples/audio_processing/audio_decoder.ipynb
Good example here

**Review by rrawther — CHANGES_REQUESTED** [review]
Added more review comments

**Review by LakshmiKumar23 — DISMISSED** [review]
@rrawther @kiritigowda functionally works. Can be merged after addressing review comments.

File: docs/examples/notebooks/audio_decoder.ipynb [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+    "audio_pipeline.build()\n",
+    "audioIterator = ROCALAudioIterator(audio_pipeline)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1bb26e37-3d99-4da4-92f6-004cb4de8c6c",
+   "metadata": {},
+   "source": [
+    "The output from the iterator includes the audio data as PyTorch tensors, the corresponding labels, and the region of interest."
```
Comment (rrawther) [inline docs/examples/notebooks/audio_decoder.ipynb:193]: please add the display for the ouput of the pipeline as it is standard in notebooks to show the final output

**Review by rrawther — CHANGES_REQUESTED** [review]
Added a comment

**Comment by swetha097** [comment]
> Added a comment

Have beautified the Notebook and added a display output. Please let me know if this needs more changes

**Comment by LakshmiKumar23** [comment]
@swetha097 are all the review comments addressed?

**Review by rrawther — APPROVED** [review]

File: docs/examples/notebooks/audio_decoder.ipynb (hunk 10 above)
Comment (swetha097) [reply docs/examples/notebooks/audio_decoder.ipynb]: addressed

File: docs/examples/notebooks/audio_decoder.ipynb (hunk 11 above)
Comment (swetha097) [reply docs/examples/notebooks/audio_decoder.ipynb]: Made modifications

**Comment by swetha097** [comment]
> @swetha097 are all the review comments addressed?

All the comments have been addressed.

**Review by LakshmiKumar23 — APPROVED** [review]

**Comment by rrawther** [comment]
This is needed for audio doc PR

**Comment by rrawther** [comment]
Accidentally closed it

**Pull request merged**